alpha/scripts/utils/ExportByEntryListToDifferentStorage.php

```diff
@@ -0,0 +1,142 @@
+		if (!$entry)
+		{
+			KalturaLog::debug("Entry not found $entryId - skipping");
+			continue;
+		}
+
+		KalturaLog::debug('Retrieving non-source assets for entry ' . $entry->getId());
+		$c = new Criteria();
+		$c->add(assetPeer::ENTRY_ID, $entry->getId());
+		$c->add(assetPeer::TYPE, array(assetType::FLAVOR), Criteria::IN);
+		$c->add(assetPeer::STATUS, flavorAsset::FLAVOR_ASSET_STATUS_READY, Criteria::IN);
+		$c->addAnd(assetPeer::IS_ORIGINAL, false);
+		$assets = assetPeer::doSelect($c);
+
+		KalturaLog::debug('Found ' . count($assets) . ' non-source assets for entry ' . $entry->getId());
+
+		foreach ($assets as /** @var flavorAsset $asset * */ $asset)
+		{
+			KalturaLog::debug('Handling asset with id ' . $asset->getId());
+			if ($externalStorage->shouldExportFlavorAsset($asset))
+			{
+				$criteria = new Criteria(FileSyncPeer::DATABASE_NAME);
+				$criteria->add(FileSyncPeer::PARTNER_ID, $asset->getPartnerId(), Criteria::EQUAL);
+				$criteria->add(FileSyncPeer::OBJECT_ID, $asset->getId(), Criteria::EQUAL);
+//				$criteria->add(FileSyncPeer::STATUS, FileSync::FILE_SYNC_STATUS_READY, Criteria::EQUAL);
+				$criteria->add(FileSyncPeer::DC, array($sourceDc, $storageId) , Criteria::IN);
+				$criteria->add(FileSyncPeer::OBJECT_TYPE, FileSyncObjectType::ASSET);
+				$criteria->add(FileSyncPeer::OBJECT_SUB_TYPE, flavorAsset::FILE_SYNC_FLAVOR_ASSET_SUB_TYPE_ASSET);
+//				$criteria->add(FileSyncPeer::FILE_PATH, 'NULL', Criteria::NOT_EQUAL);
+				$criteria->addAscendingOrderByColumn(FileSyncPeer::ID);
+				$fileSyncs = FileSyncPeer::doSelect($criteria);
+
+				$remoteDcFileSyncFound = false;
+				foreach ($fileSyncs as /** @var FileSync $fileSync * */ $fileSync)
+				{
+					if ($fileSync->getDc() === $storageId)
+					{
+						$remoteDcFileSyncFound = true;
+					}
+				}
+				if ($remoteDcFileSyncFound )
+				{
+					KalturaLog::debug("Found file sync in remote dc [$storageId] for assetId " . $asset->getId() . " . skipping exporting asset");
+					continue;
+				}
+				KalturaLog::debug("Found: " . count($fileSyncs) . " file syncs to copy");
+				foreach ($fileSyncs as /** @var FileSync $fileSync * */ $fileSync)
+				{
+					if ($fileSync->getStatus() != flavorAsset::FLAVOR_ASSET_STATUS_READY)
+					{
+						KalturaLog::debug("Found filesync " . $fileSync->getId() . " with status  " . $fileSync->getStatus() . " - skipping filesync");
+						continue;
+					}
+					try
+					{
+						KalturaLog::debug("Handling filesync " . $fileSync->getId());
+						$newfileSync = $fileSync->cloneToAnotherStorage($storageId);
+						$newfileSync->save();
+						KalturaLog::debug('New FileSync created ' . $newfileSync->getId());
+					}
+					catch (Exception $e)
+					{
+						KalturaLog::warning("Could not create newFileSync for fileSync [" . $fileSync->getId() . "]" . $e->getMessage());
+					}
+				}
+			}
+			else
+			{
+				KalturaLog::debug('Asset ' . $asset->getId() . ' should not be Exported to remote storge ' . $storageId);
+			}
```

Comment:
Upon every cycle - need to clean allocated memory.
	kMemoryManager::clearMemory();